### PR TITLE
Rabie Week1

### PR DESCRIPTION
**Pull request by rabrad**
For review

### CODE REVIEW

File: Week1/homework/createTb.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+connection.connect(err => {
+  if (err) throw err;
+  console.log('Connected...');
+});
+
+// Create tables country / city
+const tables = [
+  "CREATE TABLE IF NOT EXISTS country (ID int auto_increment primary key, Name varchar(50), Continent enum('Asia','Europe', 'North America', 'Africa','Oceania', 'Antarctica', 'South America'), Region varchar(50), SurfaceArea float, IndepYear int, Population int, LifeExpectancy float, GNP float, GNPOld float, LocalName varchar(50), GovernmentForm varchar(50), HeadOfState varchar(50), Capital varchar(50))",
+  'CREATE TABLE IF NOT EXISTS city (ID int auto_increment primary key, Name varchar(50), CountryCode varchar(50), District varchar(50), Population int)',
```
Comment (mogaleaf) [inline Week1/homework/createTb.js]: Create are good, just CountryCode is usually only three chars.

File: Week1/homework/injectValues.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    connection.query(val[i], function(err) {
+      if (err) console.log(`message: ${err}`);
+      console.log('Data has been injected into the table');
+    });
+  }
+}
+
+/*Country table model: ID, Name, Continent, Region, SurfaceArea, IndepYear, Population, LifeExpectancy, GNP, GNPOld, LocalName, GovernmentForm, HeadOfState, Capital */
+let countries = [
+  "insert into country values ( NULL ,'Syria', 'Asia', 'Middle East','185180.00','1941', '16125000', '68.5','65984.00', '64926.00', 'Suriya', 'Republic', 'MF', 'Damascus'  )",
```
Comment (mogaleaf) [inline Week1/homework/injectValues.js]: Good and autoincrement is used 👍

File: Week1/homework/injectValues.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  "insert into country values ( NULL, 'Germany', 'Europe', 'Western Europe', '357022.00', '1955', '82164700', '77.4', '2133367.00', '2102826.00', 'Deutschland', 'Federal Republic', 'Johannes Rau', 'Berlin')",
+  "insert into country values ( NULL, 'Ireland', 'Europe', 'British Islands', '70273.00', '1921', '3775100', '76.8', '75921.00', '73132.00', 'Ireland/Ã‰ire', 'Republic', 'Mary McAleese', 'Dublin')",
+  "insert into country values ( NULL, 'Spain', 'Europe', 'Southern Europe', '505992.00', '1492', '39441700', '78.8', '553233.00', '532031.00', 'Espar±a', 'Constitutional Monarchy', 'Juan Carlos I', 'Madrid')",
+  "insert into country values ( NULL, 'France', 'Europe', 'Western Europe', '551500.00', '843', '59225700', '78.8', '1424285.00', '1392448.00', 'France', 'Republic', 'Emmanuel Macron', 'Paris')",
+  "insert into country values ( NULL, 'Swaziland', 'Africa', 'Southern Africa', '17364.00', '1968', '1008000', '40.4', '1206.00', '1312.00', 'kaNgwane', 'Monarchy', 'Mswati III', 'Mbabane')",
+];
+
+//City table model: ID, Name, CountryCode, District, Population
+let cities = [
+  "insert into city values (NULL, 'Damascus', 'SYR', 'Damascus', '1347000')",
```
Comment (mogaleaf) [inline Week1/homework/injectValues.js]: Good here too !

File: Week1/homework/queries.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  connection.query(query, (err, results, fields) => {
+    if (err) throw err;
+    for (i in results) {
+      console.log(results[i]);
+    }
+  });
+}
+
+// Case4.a: What are the names of the countries with population greater than 8 million?
+const case4A = 'SELECT Name, Population FROM country WHERE Population > 8000000;';
```
Comment (mogaleaf) [inline Week1/homework/queries.js]: 👍 but no need of the ";" at the end

File: Week1/homework/queries.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+sendQuery(case4A);
+
+// Case4.b: What are the names of the countries that have “land” in their names
+const case4B = 'SELECT Name As Have_Land_In_Their_Name FROM country WHERE Name like "%land%";';
```
Comment (mogaleaf) [inline Week1/homework/queries.js]: 👍

File: Week1/homework/queries.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+sendQuery(case4B);
+
+//Case4.c: What are the names of the cities with population in between 500,000 and 1 million?
+const case4C = 'SELECT Name, Population FROM city WHERE Population BETWEEN 500000 AND 1000000;';
```
Comment (mogaleaf) [inline Week1/homework/queries.js]: 👍

File: Week1/homework/queries.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+sendQuery(case4C);
+
+//Case4.d: What are the names of all the countries on the continent ‘Europe’?
+const case4D = `SELECT Name As European_Countries FROM country WHERE Continent = 'Europe';`;
```
Comment (mogaleaf) [inline Week1/homework/queries.js]: 👍

File: Week1/homework/queries.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+sendQuery(case4D);
+
+//Case4.e: List all the countries in the descending order based on their surface areas.
+const case4E = `SELECT Name, SurfaceArea FROM country ORDER BY SurfaceArea DESC;`;
```
Comment (mogaleaf) [inline Week1/homework/queries.js]: 👍

File: Week1/homework/queries.js [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+sendQuery(case4E);
+
+//Case5.a: What are the names of all the cities in the Netherlands?
+const case5A = `SELECT Name As Netherlands_Cities FROM city WHERE CountryCode = 'NLD';`;
```
Comment (mogaleaf) [inline Week1/homework/queries.js]: 👍

File: Week1/homework/queries.js [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+sendQuery(case5A);
+
+//Case5.b: What's the population of Rotterdam?
+const case5B = `SELECT Name, Population FROM city WHERE name = 'Rotterdam';`;
```
Comment (mogaleaf) [inline Week1/homework/queries.js]: 👍

File: Week1/homework/queries.js [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+sendQuery(case5B);
+
+//Case5.c: What's the top 10 countries based on surface area?
+const case5C = `SELECT Name, SurfaceArea FROM country ORDER BY SurfaceArea DESC LIMIT 10;`;
```
Comment (mogaleaf) [inline Week1/homework/queries.js]: 👍

File: Week1/homework/queries.js [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+sendQuery(case5C);
+
+//Case5.d: What's the top 10 cities with the highest population?
+const case5D = `SELECT Name, Population FROM city ORDER BY Population DESC LIMIT 10;`;
```
Comment (mogaleaf) [inline Week1/homework/queries.js]: 👍

File: Week1/homework/queries.js [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+sendQuery(case5D);
+
+//Case5.e: What's the population of the world ?
+var case5E = 'SELECT SUM(Population) FROM country';
```
Comment (mogaleaf) [inline Week1/homework/queries.js]: 👍

File: Week1/homework/queries.js (hunk 4 above)
Comment (rabrad) [reply Week1/homework/queries.js]: > Hello,
> 
> This is a really good homework. The code is neat and all the queries are good.
> Congrats!
> You don't need to add the ';' when requesting queries with nodejs.

Hi @mogaleaf , 
Thank you so much for your kind words and review.

**Pull request closed without merging**